### ISSUE TITLE
loss average

### ISSUE THREAD

**Issue by Hannibal046** [PANXiao1994/mRASP2#9]
Hi,in this line https://github.com/PANXiao1994/mRASP2/blob/36c17003dcd642affbe8290c8f26231fec77794a/mcolt/criterions/label_smoothed_cross_entropy_with_contrastive.py#L105
why is it sum rather than mean? Does fairseq library will automatically do average in batch? Sorry , I am not familiar with this framework. And I also notice that reduce function is  sum in compute_loss https://github.com/pytorch/fairseq/blob/14c5bd027f04aae9dbb32f1bd7b34591b61af97f/fairseq/criterions/label_smoothed_cross_entropy.py#L46
and ntokens/nsenteces mean average token number within a batch, right?
https://github.com/PANXiao1994/mRASP2/blob/36c17003dcd642affbe8290c8f26231fec77794a/mcolt/criterions/label_smoothed_cross_entropy_with_contrastive.py#L66
Could you please tell the loss in the early training stage , because according to my empirical experiment, without multiplying ntokens/nsentences to contrastive_loss, it is already in the same order of magnitude, thanks so much!

**Comment by Hannibal046** [comment]
this is the contrastive loss, which is far more big than cross_entropy_loss which is typically near 10 in a randomly initialized transformer
<img width="1421" alt="image" src="https://user-images.githubusercontent.com/38466901/132002797-de40ebe0-7647-4d63-89d4-18719c89350e.png">

**Comment by PANXiao1994** [comment]
(1) Yes, it should be sum()
see cross_entropy_loss: https://github.com/pytorch/fairseq/blob/a61107ecb5c2d212b57879d5d4167030a383d82b/fairseq/criterions/label_smoothed_cross_entropy.py#L33, it is also a 'sum'
(2) " * ntokens/nsenteces" is because CE loss is token-level while CTR loss is sentence-level.
(3) Your observation of the early stages is true and reasonable. However, we should consider the magnitude of the two loss functions at the late stages, when CTR loss is very small compared with CE loss. You can try to draw the trends of them and you will observe a shaper drop of CTR loss.
